docs_user/modules/openstack-planning.adoc

```diff
@@ -386,16 +386,147 @@ addresses for the new control plane services.
 
 The general idea here is to define new IP ranges for control plane services
 that belong to a different subnet that was not used in the existing cluster.
-Then, configure IP routing between the old and new subnets to allow old and new
-service deployments to communicate.
+Then, configure link local IP routing between the old and new subnets to allow
+old and new service deployments to communicate. This involves using TripleO
+mechanism on pre-adopted cluster to configure additional link local routes
+there. This will allow EDP deployment to reach out to adopted nodes using their
+old subnet addresses.
 
 The new subnet should be sized appropriately to accommodate the new control
 plane services, but otherwise doesn't have any specific requirements as to the
-existing deployment allocation pools already consumed.
+existing deployment allocation pools already consumed. Actually, the
+requirements as to the size of the new subnet are lower than in the second
+scenario, as the old subnet ranges are kept for the adopted nodes, which means
+they don't consume any IP addresses from the new range.
+
+In this scenario, you will configure `NetworkAttachmentDefinition` CRs to use a
+different subnet from what will be configured in `NetConfig` CR for the same
+networks. The former range will be used for podified control plane services,
+while the latter will be used to manage IPAM for EDP nodes.
+
+During the process, you will need to make sure that adopted node IP addresses
+don't change during the adoption process. This is achieved by listing the
+addresses in `fixedIP` fields in `OpenstackDataplaneNodeSet` per-node section.
+
+---
+
+Before proceeding, configure host routes on the adopted nodes for the podified
+control plane subnets.
+
+To achieve this, you will need to re-run `tripleo deploy` with additional
+`routes` entries added to `network_config`. (This change should be applied
+for every adopted node configuration.) For example, you may add the following
+to `net_config.yaml`:
+
+```yaml
+network_config:
+  - type: ovs_bridge
+    name: br-ctlplane
+    routes:
+    - ip_netmask: 0.0.0.0/0
+      next_hop: 192.168.1.1
+    - ip_netmask: 172.31.0.0/24  # <- new ctlplane subnet
+      next_hop: 192.168.1.100    # <- adopted node ctlplane IP address
```

Comment:
So e.g. for setting up a route for the new storage subnet, the `next_hop` used would be the node's own IP on the old storage subnet?

Reply:
The route is slightly different (dev <bridge> in OCP; via <IP> in OSP), but the effect seems identical. The reason why this works in OSP is because Linux, when sending a reply, will pick the route with the longest matching prefix with the destination. (Without this route, the frame was sent to default gateway, which then redirected it back to destination. This of course assumed that the default gateway would redirect and should not be relied on.)

Reply:
Ack thanks for the explanation. Merging :)